### DISCUSSION TITLE
Temporarily disable history for autosuggestions

### DISCUSSION THREAD

**Discussion by oliversturm** [fish-shell/fish-shell#10597] [Q&A] [answered]
See title -- I don't think this is currently possible, or is it?

Here's why I'm asking. I often use command lines during presentations. Like with a code demo, I talk about what I'm doing while I type the command, one piece at a time. I don't want an autosuggestion to pop up, showing the entire command from a previous execution, way before I'm done! Or worse, a command that is slightly different and confuses everything...

The perfect solution would be, for me, a command that says "for this session, stop suggesting anything on the basis of history". Thoughts?

**Comment by ridiculousfish** [comment, marked as answer]
Probably the best approach is to set the environment variable `fish_history` to an empty string, e.g. `fish_history='' fish`. This will suppress loading or saving history to disk - it will still be present within the session: commands you enter during the presentation may be suggested, but historic commands will not be.

If you want to prevent all commands from history from being suggested, probably the best way is to clear the history after every command:

```sh
function clear_history --on-event fish_postexec
    builtin history clear-session
end
```

this will prevent up-arrow from recalling commands, but it should otherwise have the intended effect.
